From 075e801e25c6aa31b220a5d89246a9fb73bbe578 Mon Sep 17 00:00:00 2001
From: Ajay Kumar Gupta <ajay.gupta@ti.com>
Date: Thu, 25 Feb 2010 17:43:29 +0530
Subject: [PATCH 010/108] DM37x: musb: Fix core off issue

commit 1f410f2a3f3f1394d0c9d416cabe3ddbeb23ee15 from
the branch origin/OMAPPSP_03.00.01.06 of
git://arago-project.org/git/projects/linux-omap3.git

Enable FORCEIDLE for musb as SMARTIDLE is blocking the core to enter
into off mode when off mode flags are enabled.

Signed-off-by: Ajay Kumar Gupta <ajay.gupta@ti.com>
Integrated-by: Stanley.Miao <stanley.miao@windriver.com>
---
 drivers/usb/musb/omap2430.c |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/drivers/usb/musb/omap2430.c b/drivers/usb/musb/omap2430.c
index f77a21a..aad2b41 100644
--- a/drivers/usb/musb/omap2430.c
+++ b/drivers/usb/musb/omap2430.c
@@ -225,7 +225,12 @@ int __init musb_platform_init(struct musb *musb)
 	l |= SMARTSTDBY;	/* enable smart standby */
 	l &= ~AUTOIDLE;		/* disable auto idle */
 	l &= ~NOIDLE;		/* remove possible noidle */
-	l |= SMARTIDLE;		/* enable smart idle */
+
+	/* SMARTIDLE is blocking core to enter off mode in 3630 */
+	if (cpu_is_omap3630())
+		l |= FORCEIDLE;		/* enable force idle */
+	else
+		l |= SMARTIDLE;		/* enable smart idle */
 	/*
 	 * MUSB AUTOIDLE don't work in 3430.
 	 * Workaround by Richard Woodruff/TI
-- 
1.7.0.4

